ads/google/a4a/test/test-traffic-experiments.js

```diff
@@ -346,4 +370,219 @@ describe('all-traffic-experiments-tests', () => {
       expect(isInExperiment(element, 'zort')).to.be.true;
     });
   });
+
+  describe('A4A Launch Flags', () => {
+    let sandbox;
+    let win;
+    let rand;
+    let events;
+    let element;
+
+    beforeEach(() => {
+      sandbox = sinon.sandbox.create();
+      rand = sandbox.stub(Math, 'random');
+      win = {
+        AMP_MODE: {
+          localDev: true,
+        },
+        location: {
+          href: 'https://cdn.ampproject.org/fnord',
+          pathname: '/fnord',
+          origin: 'https://cdn.ampproject.org',
+          hash: '',
+          hostname: 'cdn.ampproject.org',
+        },
+        document: {
+          nodeType: /* DOCUMENT */ 9,
+          hidden: false,
+          cookie: null,
+          visibilityState: 'visible',
+          addEventListener: function(type, listener) {
+            events[type] = listener;
+          },
+        },
+        crypto: {
+          subtle: true,
+          webkitSubtle: true,
+        },
+        navigator: window.navigator,
+        pageExperimentBranches: {},
+      };
+      win.document.defaultView = win;
+      installDocService(win, /* isSingleDoc */ true);
+      const ampdoc = ampdocServiceFor(win).getAmpDoc();
+      events = {};
+      installDocumentStateService(win);
+      installPlatformService(win);
+      installViewerServiceForDoc(ampdoc);
+      element = document.createElement('div');
+      element.setAttribute('type', 'doubleclick');
+      document.body.appendChild(element);
+      toggleExperiment(win, EXP_ID, true, true);
+    });
+
+    afterEach(() => {
+      resetServiceForTesting(win, 'viewer');
+      sandbox.restore();
+      document.body.removeChild(element);
+    });
+
+    function checkAgainstOtherBranches(element, eid) {
+      const branchIds = [
+        DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH.control,
+        DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH.experiment,
+        DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH.control,
+        DOUBLECLICK_A4A_EXTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH.experiment,
+        DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH.control,
+        DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_PRE_LAUNCH.experiment,
+        DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH.control,
+        DOUBLECLICK_A4A_INTERNAL_EXPERIMENT_BRANCHES_POST_LAUNCH.experiment,
+      ];
+      for (const branchId in branchIds) {
+        if (branchId === eid) {
```

Comment:
Could make this `expect(isInExperiment(element, branchId)).to.equal(branchId === eid)`.